extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/security/HttpSecurity.java

```diff
@@ -78,7 +78,31 @@
 public interface HttpSecurity {
 
     /**
-     * Creates {@link HttpPermission}.
+     * Registers the Basic authentication mechanism in addition to all other global authentication mechanisms.
+     *
+     * @param basic {@link Basic} instance that replaces the configuration in the 'application.properties' file
+     * @return HttpSecurity
+     */
+    HttpSecurity mechanism(Basic basic);
```

Comment:
Hmm, my earlier comment was assuming it was `mechanism(HttpAuthenticationMechanism)`

Reply:
That can be done easily, I have following problem with that approach: what if users configure something in the `application.properties` and then do Basic.realm() and it builds `HttpAuthenticationMechanism`? Which configuration are we going to keep? Are we going to detect `BasicAuthMech` instance and don't add the original mechanism? So `Basic` and `Form` are just ways to configure form/basic, they don't built actual mechanism.

Reply:
Done, there is validation in place and 2 tests: `FluentApiBasicAuthValidationFailureTest` and `FluentApiFormAuthValidationFailureTest`